multiverse/approvalweight_manager.go

```diff
@@ -38,14 +42,14 @@ func (a *ApprovalManager) ApproveMessages(messageID MessageID) {
 	weight := a.tangle.WeightDistribution.Weight(issuingMessage.Issuer)
 	a.tangle.Utils.WalkMessagesAndMetadata(func(message *Message, messageMetadata *MessageMetadata, walker *walker.Walker) {
 
-		weightByte := message.WeightSlice[int(byteIndex)]
-		if weightByte&(1<<mod) != 0 {
+		weightByte := messageMetadata.weightSlice[int(byteIndex)]
+		if weightByte&(1<<mod) == 0 {
 			weightByte |= 1 << mod
-			message.Weight += weight
-			a.Events.MessageWeightUpdated.Trigger(message.ID)
-			if float64(message.Weight) >= config.MessageWeightThreshold*float64(a.tangle.WeightDistribution.TotalWeight()) {
-				message.ConfirmationTime = time.Now()
-				a.Events.MessageConfirmed.Trigger(a.tangle.Peer.ID, message.Issuer, message.ID, message.IssuanceTime, message.ConfirmationTime, message.Weight)
+			messageMetadata.weight += weight
+			a.Events.MessageWeightUpdated.Trigger(message, messageMetadata, messageMetadata.weight)
+			if float64(messageMetadata.weight) >= config.MessageWeightThreshold*float64(a.tangle.WeightDistribution.TotalWeight()) {
```

Comment:
Will the confirmation time of the confirmed messages updated again and again?

Reply:
fixed